crypto/aes/aes_core.c

```diff
@@ -50,7 +50,577 @@
 #include <openssl/aes.h>
 #include "aes_local.h"
 
-#ifndef AES_ASM
+#if defined(OPENSSL_AES_CONST_TIME) && !defined(AES_ASM)
+typedef union
+{
+    unsigned char b[8];
+    u32 w[2];
+    u64 d;
+} uni;
+
+static void XtimeWord(u32 *w)
+{
+    u32 a, b;
+    a = *w;
+    b = a & 0x80808080u;
+    a ^= b;
+    b -= b >> 7;
+    b &= 0x1B1B1B1Bu;
+    b ^= a << 1;
+    *w = b;
+}
+
+static void XtimeLong(u64 *w)
+{
+    u64 a, b;
+    a = *w;
+    b = a & 0x8080808080808080uLL;
+    a ^= b;
+    b -= b >> 7;
+    b &= 0x1B1B1B1B1B1B1B1BuLL;
+    b ^= a << 1;
+    *w = b;
+}
+
+static void SubWord(u32 *w)
+{
+    u32 x, y, a1, a2, a3, a4, a5, a6;
+    x = *w;
+    y = ((x & 0xFEFEFEFEu) >> 1) | ((x & 0x01010101u) << 7);
```

Comment:
Would it be worthwhile making this into a macro with a descriptive name.
I find code like this hard to read -- my eyes head straight for the big constants.
```
#define bytewise_rotate_right_one_bit(n) ((n & 0xFEFEFEFEu) >> 1) | ((n & 0x01010101u) << 7)
```

Reply:
better shut your eyes here.